### PR TITLE
Containerize gocounter example

### PR DESCRIPTION
**Pull request by Billy99**
Containerize the gocounter example. Includes the following changes:
* Add Containerfile.gocounter to allow building gocounter userspace program in a container.
* Add Containerfile.gocounter-build to allow building gocounter bytecode and gocounter userspace program from a container instead of on the host.
* gocounter bytecode in container image has been pushed to quay.io/bpfd/bytecode:gocounter.
* Update gocounter to take parametertized input using go flags.
* Update gocounter main to accept a different file location for bytecode (was hardcoded), a URL of the bytecode or a bpfd UUID of a preloaded bytecode.
* Update gocounter to take parameters from a .toml file.
* Add a gocounter-entrypoint.sh for gocounter running in a container. This entrypoint script takes gocounter parameters passed in as environment variables.
* Add gocounter-bytecode.yaml and gocounter.yaml to deploy on a k8s deployment.
* Beef-up documentation of go.md with images and multiple deployments.

Two other bug fixes that were encountered:
1) Code panic'd if there was an issue unpacking the bytecode from a container image.
```
Sep 22 16:10:22 ebpf03 bpfd[424634]: thread 'tokio-runtime-worker' panicked at 'called `Result::unwrap()` on an `Err` value: Custom { kind: PermissionDenied, error: TarError { desc: "failed to unpack `/var/bpfd/bytecode/bpf_bpfel.o`", io: Custom { kind: PermissionDenied, error: TarError { desc: "failed to unpack `bpf_bpfel.o` into `/var/bpfd/bytecode/bpf_bpfel.o`", io: Os { code: 13, kind: PermissionDenied, message: "Permission denied" } } } } }', bpfd/src/server/pull_bytecode.rs:93:49
```

2) When running as a systemd service, loading from a URL fails because `/var/bpfd/bytecode/` is owned by `root:bpfd`, and the service is running user `bpfd`:
```
$ bpfctl load --iface vethb2795c7 --priority 50 --from-image quay.io/bpfd/bytecode:gocounter
Error: status: Aborted, message: "failed to unpack `/var/bpfd/bytecode/bpf_bpfel.o`", details: [], metadata: MetadataMap { headers: {"content-type": "application/grpc", "date": "Mon, 03 Oct 2022 13:03:07 GMT", "content-length": "0"} }
```
Update user.sh to make sure /var/bpfd/bytecode/` is owned by `bpfd:bpfd`.
With the above change, when `./scripts.sh install` or `./scripts.sh init` is run, `/var/bpfd/bytecode/` is owned by `bpfd:bpfd` as expected. If bpfd is then start from binary instead of systemd service, it is being run as `root:bpfd`, and spawned
threads which are stripped of capabilites can't load from URL:
```
$ sudo ./target/debug/bpfctl load --iface vethb2795c7 --priority 50 --from-image quay.io/bpfd/bytecode:gocounter
Error: status: Aborted, message: "failed to unpack `/var/bpfd/bytecode/bpf_bpfel.o`", details: [], metadata: MetadataMap { headers: {"content-type": "application/grpc", "date": "Mon, 03 Oct 2022 14:19:50 GMT", "content-length": "0"} }
```
Update bpfd main.rs to only drop all capabilities on spawned threads if current user is bpfd. This is a short term solution and needs to be revisited with #166.

### CODE REVIEW

**Comment by astoycos** [comment]
Thanks for getting to this! I'll get to reviewing it ASAP :)

**Comment by Billy99** [comment]
> Thanks for getting to this! I'll get to reviewing it ASAP :)

Feel free to start reviewing, working through an integration test failure at the moment.

File: packaging/kubernetes-deployment/sample-ebpfprograms/gocounter-bytecode.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+apiVersion: bpfd.io/v1alpha1
```
Comment (Billy99) [inline packaging/kubernetes-deployment/sample-ebpfprograms/gocounter-bytecode.yaml:1]: @astoycos Is there a way to specific which node to load program on? What if we want a different interface for each node (like load on the nginx interface, which would be different on each node)?

**Review by astoycos — CHANGES_REQUESTED** [review]
You went above and beyond here, thanks! @Billy99 

The only thing I want to keep in mind here is that "our vision" for running bpf in K8s is squarely focused on using the "bpfdagent/operator" to do so.  These are some great docs but let's make sure to specify that these are "Manual and advanced" instructions for doing so, just to help not confuse folks.

File: packaging/kubernetes-deployment/sample-ebpfprograms/gocounter-bytecode.yaml (hunk 1 above)
Comment (astoycos) [reply packaging/kubernetes-deployment/sample-ebpfprograms/gocounter-bytecode.yaml:1]: Not Yet, but I'm hoping to add that functionality ASAP

File: docs/admin/shipping-bytecode.md [hunk 2]
```diff
@@ -57,8 +57,21 @@ To use the provided templated Containerfile simply run a `podman build` command
...
+ --build-arg PROGRAM_TYPE=xdp \
+ --build-arg BYTECODE_FILENAME=pass.bpf.o \
+ --build-arg KERNEL_COMPILE_VER=$(uname -r) \
+ -f packaging/container-deployment/Containerfile.bytecode \
+ /home/<USER>/bytecode -t quay.io/<USER>/xdp_pass:latest
 ```
+Where `/home/<USER>/bytecode` is the directory the bytecode object file is located.
+
+
+**NOTE:** By default, `podman` will use the OCI media type as described above.
```
[the remark this answers is not in the document]
Comment (Billy99) [reply docs/admin/shipping-bytecode.md:72]: I am using Docker version 20.10.18 and I couldn't get it to work. I had podman on my laptop and it worked.

[the remark this answers is not in the document]
Comment (Billy99) [reply docs/admin/shipping-bytecode.md:72]:
```
$ skopeo inspect --raw docker://quay.io/billy99/gocounter-bytecode:latest
{
   "schemaVersion": 2,
   "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
   "config": {
      "mediaType": "application/vnd.docker.container.image.v1+json",
      "size": 2998,
      "digest": "sha256:e9a5646e514160b53af156f8c1c623c6469db25ecb5ba78083f4a0a1368ef3d3"
   },
   "layers": [
      {
         "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
         "size": 2642,
         "digest": "sha256:e3712661920c77635cdab75372e4f331c7cf4b7677355ff2992dfd055e37aca0"
      }
   ]
}
```

[the remark this answers is not in the document]
Comment (astoycos) [reply docs/admin/shipping-bytecode.md:72]: Huh that's weird! it's using it's own Media type there... I was running docker 1.19 but let me try again

[the remark this answers is not in the document]
Comment (astoycos) [reply docs/admin/shipping-bytecode.md:72]: I'm seeing the same thing...

https://docs.docker.com/registry/spec/manifest-v2-2/

[the remark this answers is not in the document]
Comment (astoycos) [reply docs/admin/shipping-bytecode.md:72]: We can change bpfd to account for this since it's still just a gzipped Tar, but It's pretty annoying they're not using the media types from https://github.com/opencontainers/image-spec/blob/main/media-types.md

[the remark this answers is not in the document]
Comment (astoycos) [reply docs/admin/shipping-bytecode.md:72]: Lol https://github.com/containers/skopeo/issues/1692

[the remark this answers is not in the document]
Comment (astoycos) [reply docs/admin/shipping-bytecode.md:72]: It's all good though I guess https://github.com/opencontainers/image-spec/blob/main/media-types.md#applicationvndociimagelayerv1targzip 

As it's explicitly labeled as compatible in the spec

[the remark this answers is not in the document]
Comment (astoycos) [reply docs/admin/shipping-bytecode.md:72]: I'll fix that now, but I think you can remove that note still :)

File: examples/gocounter/gocounter-entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
```
[the remark this answers is not in the document]
Comment (Billy99) [reply examples/gocounter/gocounter-entrypoint.sh:1]: I originally created an entrypoint script to copy bpf_bpfel.o out of the container and onto the host. Then I added the environment variable parsing to the script, then removed the copy of bpf_bpfel.o. I can move the environment variable parsing to main.go.

File: docs/admin/shipping-bytecode.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (Billy99) [reply docs/admin/shipping-bytecode.md:72]: Fixed on next PR.

File: docs/developer/go.md [hunk 4]
```diff
@@ -1,8 +1,77 @@
 # Go
 
-An example application that uses the `bpfd-go` bindings can be found [here](https://github.com/redhat-et/bpfd/tree/main/examples/gocounter)
+An example application that uses the `bpfd-go` bindings can be found in
+[examples/gocounter/](https://github.com/redhat-et/bpfd/tree/main/examples/gocounter).
+This example and associated documentation is intended to provide basics on deploying a BPF
+program using bpfd.
```
[the remark this answers is not in the document]
Comment (Billy99) [reply docs/developer/go.md:6]: Fixed on next PR.

File: docs/developer/go.md [hunk 5]
```diff
@@ -1,8 +1,77 @@
...
+As one will see, there a couple of ways to deploy, starting with simply running locally, and
```
[the remark this answers is not in the document]
Comment (Billy99) [reply docs/developer/go.md:7]: Fixed on next PR.

File: docs/developer/go.md [hunk 6]
```diff
@@ -1,8 +1,77 @@
...
+ending up with deploying on a Kubernetes cluster.
+
+The `gocounter` example contains a portion written in C
```
[the remark this answers is not in the document]
Comment (Billy99) [reply docs/developer/go.md:10]: Fixed on next PR.

File: docs/developer/go.md [hunk 7]
```diff
@@ -1,8 +1,77 @@
...
+([xdp_counter.c](https://github.com/redhat-et/bpfd/tree/main/examples/gocounter/bpf/xdp_counter.c))
+that is compiled into BPF bytecode.
+The BPF program counts packets that are received on an interface and stores the packet and byte
+counts in a map that is accessible by the userspace portion.
+The `gocounter` example also has a userspace portion written in GO that is responsible for loading
+the BPF program at the XDP hook point, via gRPC call to `bpfd`, and then every 3 seconds polls the
```
[the remark this answers is not in the document]
Comment (Billy99) [reply docs/developer/go.md:16]: Fixed on next PR.

File: docs/developer/go.md [hunk 8]
```diff
@@ -21,126 +90,506 @@ An example application that uses the `bpfd-go` bindings can be found [here](http
...
-    cd examples/gocounter && go generate
-```
+In this example, when `gocounter` is started, it will send a gRPC request over mTLS to `bpfd`
+requesting `bpfd` to load the gocounter BPF bytecode located at
+`src/bpfd/src/examples/gocounter/bpf_bpfel.o` at a priority of 50 and on interface `ens3`.
+These values are configurable as we will see later, but for now we will use the defaults
+(except interface, which is required to be entered).
+`bpfd` will load it's `dispatcher` BPF program, which links to the `gocounter` BPF program
+which was requested to be loaded.
+`bpfctl list` can be used to show that the BPF bytecode was loaded.
```
[the remark this answers is not in the document]
Comment (Billy99) [reply docs/developer/go.md:127]: Fixed on next PR.

File: docs/developer/go.md [hunk 9]
```diff
@@ -21,126 +90,506 @@ An example application that uses the `bpfd-go` bindings can be found [here](http
...
 
-^CExiting...
+Running the `gocounter` example in a container can take many forms.
+This will step through the different options and lay the groundwork to running the application on
+Kubernetes.
+
+### Passing BPF Bytecode In A Container Image
+
+Whether or not the Userspace portion of `gocounter` is running on a host or in a container, the
+BPF bytecode can be loaded from a container image.
```
[the remark this answers is not in the document]
Comment (Billy99) [reply docs/developer/go.md:272]: Fixed on next PR.

File: docs/developer/go.md [hunk 10]
```diff
@@ -21,126 +90,506 @@ An example application that uses the `bpfd-go` bindings can be found [here](http
...
+
+This section will describe loading bytecode on a Kubernetes cluster and launching the Userspace
+program.
+The approach is slightly different when running on a Kubernetes cluster.
+The BPF bytecode should be loaded by an administrator, not the Userspace program itself.
+
+![gocounter On Kubernetes](../img/gocounter-in-k8s.png)
+
+
+### Preloading gocounter BPF Bytecode
```
[the remark this answers is not in the document]
Comment (Billy99) [reply docs/developer/go.md:395]: This is not in K8s but on the host. I can move it up a section (out of **Deploying On Kubernetes**), but was showing that you can run gocounter via UUID (new).

File: docs/developer/go.md [hunk 11]
```diff
@@ -21,126 +90,506 @@ An example application that uses the `bpfd-go` bindings can be found [here](http
 
     `go install github.com/cilium/ebpf/cmd/bpf2go@master`
 
-## Generate certificates for the gocounter bpfd client
 
-`bpfd` uses mTLS for mutual authentication. To generate a client certificate for
-the gocounter example run:
+#### Building Locally
```
[the remark this answers is not in the document]
Comment (astoycos) [reply docs/developer/go.md:94]: But we could also do this in a follow up since this is already big

File: examples/gocounter/gocounter-entrypoint.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (astoycos) [reply examples/gocounter/gocounter-entrypoint.sh:1]: I think that'd be better if it's not too much of a PITA!

File: docs/developer/go.md [hunk 12]
```diff
@@ -1,8 +1,77 @@
...
+ending up with deploying on a Kubernetes cluster.
+
+The `gocounter` example contains a portion written in C
+([xdp_counter.c](https://github.com/redhat-et/bpfd/tree/main/examples/gocounter/bpf/xdp_counter.c))
+that is compiled into BPF bytecode.
+The BPF program counts packets that are received on an interface and stores the packet and byte
+counts in a map that is accessible by the userspace portion.
+The `gocounter` example also has a userspace portion written in GO that is responsible for loading
+the BPF program at the XDP hook point, via gRPC call to `bpfd`, and then every 3 seconds polls the
+map shared with the BPF program and logs the current counts. The `gocounter` userspace code is
```
[the remark this answers is not in the document]
Comment (Billy99) [reply docs/developer/go.md:17]: Fixed on next PR.

File: docs/developer/go.md [hunk 13]
```diff
@@ -21,126 +90,506 @@ An example application that uses the `bpfd-go` bindings can be found [here](http
...
+```console
+    kubectl get EbpfPrograms -A
+    NAMESPACE     NAME                  AGE
+    kube-system   gocounter             9s
+
+    kubectl describe EbpfPrograms -n kube-system gocounter
+    Name:         gocounter
+    Namespace:    kube-system
+    Labels:       <none>
+    Annotations:  ebpf-k8s-1/attach_point: ens3
```
[the remark this answers is not in the document]
Comment (Billy99) [reply docs/developer/go.md:474]: +1

File: docs/developer/go.md [hunk 14]
```diff
@@ -21,126 +90,506 @@ An example application that uses the `bpfd-go` bindings can be found [here](http
...
+
+```console
+    cd src/bpfd/
+    podman build -f examples/gocounter/Containerfile.gocounter . -t quay.io/$USER/gocounter:latest
+
+    podman login quay.io
+    podman push quay.io/$USER/gocounter:latest
+```
+
+Using the UUID from annotation collected in the previous section, update the
```
[the remark this answers is not in the document]
Comment (Billy99) [reply docs/developer/go.md:502]: Or update this section once the "CRD per Node" stuff goes in and make this more usable.

File: examples/gocounter/gocounter-entrypoint.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (Billy99) [reply examples/gocounter/gocounter-entrypoint.sh:1]: It is definitely easier in a script, but I can move it.

File: packaging/container-deployment/Containerfile.bpfd [hunk 15]
```diff
@@ -12,6 +10,9 @@ RUN apt-get update && apt-get install -y\
  gcc-multilib\
  musl-tools
 
+WORKDIR /usr/src/bpfd
```
[the remark this answers is not in the document]
Comment (Billy99) [reply packaging/container-deployment/Containerfile.bpfd:13]: I like to install first, then do local work. That way when code changes and you rebuild, install layer is run from cache.

File: packaging/kubernetes-deployment/sample-ebpfprograms/gocounter.yaml [hunk 16]
```diff
@@ -0,0 +1,91 @@
...
+  hostNetwork: true
+  dnsPolicy: ClusterFirstWithHostNet
+  terminationGracePeriodSeconds: 15
+  nodeSelector:
+    kubernetes.io/hostname: ebpf-k8s-2
+  containers:
+  - name: gocounter
+    image: quay.io/$USER/gocounter:latest
+    imagePullPolicy: IfNotPresent
+    #securityContext:
```
[the remark this answers is not in the document]
Comment (Billy99) [reply packaging/kubernetes-deployment/sample-ebpfprograms/gocounter.yaml:50]: I couldn't get it to work manually because docker doesn't know about CAP_BPF. I was tracing the patch but it wasn't in my version. It should work this way once docker knows about CAP_BPF. Need to test on K8s but got side tracked. So left by accident but should be done this way long term.

File: docs/developer/go.md [hunk 17]
```diff
@@ -1,8 +1,77 @@
 # Go
 
-An example application that uses the `bpfd-go` bindings can be found [here](https://github.com/redhat-et/bpfd/tree/main/examples/gocounter)
+An example application that uses the `bpfd-go` bindings can be found in
+[examples/gocounter/](https://github.com/redhat-et/bpfd/tree/main/examples/gocounter).
+This example and associated documentation is intended to provide basics on deploying a BPF
+program using bpfd.
+As one will see, there a couple of ways to deploy, starting with simply running locally, and
+ending up with deploying on a Kubernetes cluster.
```
[the remark this answers is not in the document]
Comment (Billy99) [reply docs/developer/go.md:8]: I do use the `EbpfProgram` CRD for the k8s deployment, so what am I missing?

File: packaging/container-deployment/Containerfile.bpfd (hunk 15 above)
[the remark this answers is not in the document]
Comment (astoycos) [reply packaging/container-deployment/Containerfile.bpfd:13]: Sweet that actually sounds like a nice development time saver :) 

Can you update the images for bpfctl and bpfd-agent in https://github.com/redhat-et/bpfd/tree/main/packaging/container-deployment as well just to keep everything consistent?

File: docs/developer/go.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (astoycos) [reply docs/developer/go.md:395]: Ok cool yeah that would make more sense I think! 

In that diagram we might want to show kubectl as talking to the API server, and bpfd-agent as talking to bpfd just so folks don't think kubectl is actually speaking directly to bpfd :)

File: packaging/kubernetes-deployment/sample-ebpfprograms/gocounter.yaml (hunk 16 above)
[the remark this answers is not in the document]
Comment (astoycos) [reply packaging/kubernetes-deployment/sample-ebpfprograms/gocounter.yaml:50]: Ok cool maybe just add an issue to investigate this/ perms in K8s in general and remove this for now?

File: packaging/container-deployment/Containerfile.bpfd (hunk 15 above)
[the remark this answers is not in the document]
Comment (Billy99) [reply packaging/container-deployment/Containerfile.bpfd:13]: Fixed in the next PR.

I was in Containerfile.bpfd because I was having trouble deploying and I moved it from scratch to an alpine base image so I could break into and poke around. I made the above change as well. Got it working and backed out the other changes. I think this went in by accident, but I do think it is a time saver.

File: docs/developer/go.md (hunk 17 above)
[the remark this answers is not in the document]
Comment (astoycos) [reply docs/developer/go.md:8]: I was mixxed up with https://github.com/redhat-et/bpfd/pull/178#discussion_r986165835 but moving that up out of the kube section should clear things up

File: docs/admin/shipping-bytecode.md [hunk 18]
```diff
@@ -57,8 +57,21 @@ To use the provided templated Containerfile simply run a `podman build` command
 like the following:
 
 ```bash
-podman build  --build-arg PROGRAM_NAME=xdp_counter --build-arg SECTION_NAME=pass --build-arg PROGRAM_TYPE=xdp --build-arg BYTECODE_FILENAME=pass.bpf.o --build-arg KERNEL_COMPILE_VER=$(uname -r) -f packaging/container-deployment/Containerfile.bytecode /home/<USER>/bytecode -t quay.io/<USER>/xdp_pass:latest
+podman build \
+ --build-arg PROGRAM_NAME=xdp_counter \
```
[the remark this answers is not in the document]
Comment (astoycos) [reply docs/admin/shipping-bytecode.md:61]: this is already fixed

File: docs/developer/go.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (Billy99) [reply docs/developer/go.md:94]: +1, deferring

File: examples/gocounter/gocounter-entrypoint.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (Billy99) [reply examples/gocounter/gocounter-entrypoint.sh:1]: Fixed on next PR.

File: examples/gocounter/main.go [hunk 19]
```diff
@@ -26,67 +30,210 @@ type Stats struct {
...
-	if err != nil {
-		log.Fatalf("did not connect: %v", err)
-	}
-	c := gobpfd.NewLoaderClient(conn)
+	flag.StringVar(&iface, "iface", "",
+		"Interface to load bytecode.")
+	flag.IntVar(&priority, "priority", -1,
+		"Priority to load program in bpfd")
+	flag.StringVar(&bytecodeUuid, "uuid", "",
+		"UUID of bytecode that has already been loaded. uuid, url and path are mutually exclusive.")
```
[the remark this answers is not in the document]
Comment (Billy99) [reply examples/gocounter/main.go:86]: Maybe later

File: packaging/kubernetes-deployment/sample-ebpfprograms/gocounter.yaml [hunk 20]
```diff
@@ -0,0 +1,91 @@
+---
```
[the remark this answers is not in the document]
Comment (Billy99) [reply packaging/kubernetes-deployment/sample-ebpfprograms/gocounter.yaml:1]: Fixed in the next PR.

Moved `gocounter.yaml` and `gocounter-bytecode.yaml` to `examples/gocounter/kubernetes-deployment/.`
Also moved `Containerfile.gocounter` to the subdirectory `examples/gocounter/container-deployment/.`

File: packaging/kubernetes-deployment/sample-ebpfprograms/gocounter.yaml (hunk 16 above)
[the remark this answers is not in the document]
Comment (Billy99) [reply packaging/kubernetes-deployment/sample-ebpfprograms/gocounter.yaml:50]: Removed for now.

File: examples/gocounter/Containerfile.gocounter-build [hunk 21]
```diff
@@ -0,0 +1,16 @@
+FROM golang:1.19
```
[the remark this answers is not in the document]
Comment (Billy99) [reply examples/gocounter/Containerfile.gocounter-build:1]: I forget why I added. I think in a meeting there was a comment about making it easy for user and minimize impact. I'll remove.

File: docs/developer/go.md (hunk 17 above)
[the remark this answers is not in the document]
Comment (Billy99) [reply docs/developer/go.md:8]: Done

File: docs/developer/go.md [hunk 22]
```diff
@@ -1,8 +1,77 @@
...
+- [Deploying On Kubernetes](#deploying-on-kubernetes)
+   - [Preloading gocounter BPF Bytecode](#preloading-gocounter-bpf-bytecode)
+   - [Loading gocounter BPF Bytecode On Kubernetes](#loading-gocounter-bpf-bytecode-on-kubernetes)
+- [Notes](#notes)
 
-## Prerequisites
+## Building
+
+There are two ways to build the example code.
+One way is to use the provided Containerfile to build a Build-Container and then use that
```
[the remark this answers is not in the document]
Comment (Billy99) [reply docs/developer/go.md:42]: Removed.

**Comment by Billy99** [comment]
@astoycos I think I addressed all your comments. Let me know if I missed any

File: examples/gocounter/container-deployment/Containerfile.gocounter [hunk 23]
```diff
@@ -0,0 +1,25 @@
...
+
+WORKDIR /usr/src/bpfd/
+COPY ./ /usr/src/bpfd/
+
+WORKDIR /usr/src/bpfd/examples/gocounter
+
+# Compile gocounter
+RUN go generate
+RUN go build
+RUN /sbin/setcap cap_dac_override,cap_bpf=ep ./gocounter
```
Comment (astoycos) [inline examples/gocounter/container-deployment/Containerfile.gocounter:19]: Is this still supposed to be there?

**Comment by astoycos** [comment]
One small question but otherwise this /lgtm, thanks @Billy99!

File: docs/developer/go.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (astoycos) [reply docs/developer/go.md]: Lets make an issue for this? Since I think it will be subtly important/ we can point people to this example

[the remark this answers is not in the document]
Comment (Billy99) [reply docs/developer/go.md]: https://github.com/redhat-et/bpfd/issues/187

File: examples/gocounter/container-deployment/Containerfile.gocounter (hunk 23 above)
Comment (Billy99) [reply examples/gocounter/container-deployment/Containerfile.gocounter]: Removed.
As part of this change and retesting, found that I couldn't get it run with a base image of `scratch` or `alpine`. Keep getting the error `exec ./gocounter: no such file or directory`. With one of the runs (can't remember the base image, might have been alpine) I got a little more info:
```
2022/10/07 18:19:22 Program registered with af263886-ce19-4657-8275-d51b3e545fd0 id
2022/10/07 18:19:22 Failed to load pinned Map: /var/run/bpfd/fs/maps/af263886-ce19-4657-8275-d51b3e545fd0/xdp_stats_map
2022/10/07 18:19:22 no such file or directory
```
So it has to do with mounting the map. I was able to get `registry.fedoraproject.org/fedora-minimal:latest` to work, so used that.

Comment (astoycos) [reply examples/gocounter/container-deployment/Containerfile.gocounter]: Hrm that's strange... I'm fine with minimal, in the future we should explore UBI rather than scratch

**Review by astoycos — APPROVED** [review]
